python/pyspark/sql/connect/dataframe.py

```diff
@@ -124,6 +125,29 @@ def withPlan(cls, plan: plan.LogicalPlan, session: "RemoteSparkSession") -> "Dat
     def select(self, *cols: "ExpressionOrString") -> "DataFrame":
         return DataFrame.withPlan(plan.Project(self._plan, *cols), session=self._session)
 
+    def selectExpr(self, *expr: Union[str, List[str]]) -> "DataFrame":
+        """Projects a set of SQL expressions and returns a new :class:`DataFrame`.
+
+        This is a variant of :func:`select` that accepts SQL expressions.
+
+        .. versionadded:: 3.4.0
+
+        Returns
+        -------
+        :class:`DataFrame`
+            A DataFrame with new/old columns transformed by expressions.
+        """
+        sql_expr = []
+        if len(expr) == 1 and isinstance(expr[0], list):
+            expr = expr[0]  # type: ignore[assignment]
+        for element in expr:
+            if isinstance(element, str):
+                sql_expr.append(SQLExpression(element))
+            else:
+                sql_expr.extend([SQLExpression(e) for e in element])
+
+        return DataFrame.withPlan(plan.Project(self._plan, *sql_expr), session=self._session)
```

Comment:
So this becomes an unresolved attribute and just works out of the box?

Reply:
Actually it becomes `Expression.Builder().setExpressionString()` which are SQL expression strings.

`str` could be different things in DataFrame API.

Reply:
This was not necessarily the question that I had, but I was not remembering correctly the type interface to Project:

```
def __init__(self, child: Optional["LogicalPlan"], *columns: "ExpressionOrString") -> None:
```

In this case `SQLExpression` is an expression and it just works.